waku/v2/protocol/waku_relay.nim

```diff
@@ -100,25 +96,28 @@ proc new*(T: type WakuRelay,
       verifySignature = false,
       maxMessageSize = MaxWakuMessageSize
     )
-
-    # Rejects messages that are not WakuMessage
-    proc validator(topic: string, message: messages.Message): Future[ValidationResult] {.async.} =
-      let msg = WakuMessage.decode(message.data)
-      if msg.isOk():
-        return ValidationResult.Accept
-      return ValidationResult.Reject
-
-    # Add validator to all default pubsub topics
-    for pubSubTopic in defaultPubsubTopics:
-      wr.addValidator(pubSubTopic, validator)
   except InitializationError:
     return err("initialization error: " & getCurrentExceptionMsg())
 
-  wr.defaultPubsubTopics = defaultPubsubTopics
+    # TODO: Add a function to validate the WakuMessage integrity
+    # # Rejects messages that are not WakuMessage
+    # proc validator(topic: string, message: messages.Message): Future[ValidationResult] {.async.} =
+    #   let msg = WakuMessage.decode(message.data)
+    #   if msg.isOk():
+    #     return ValidationResult.Accept
+    #   return ValidationResult.Reject
+
+    # # Add validator to all default pubsub topics
+    # for pubSubTopic in defaultPubsubTopics:
+    #   wr.addValidator(pubSubTopic, validator)
```

Comment:
I still don't see the problem with this. imho its better an existing feature that solves a real problem (with room to improve) than a non existing feature.

The initial concerns were
> The Waku message integrity validator is only being applied to the default pub-sub topics ([here](https://github.com/waku-org/nwaku/blob/master/waku/v2/protocol/waku_relay.nim#L105-L113)) since the validator is only applied at the WakuRelay instance creation.

Dont see a problem here, was the intention of the PR.

> If any user subscribes to a pub-sub topic via the subscribe API ([here](https://github.com/waku-org/nwaku/blob/master/waku/v2/protocol/waku_relay.nim#L138-L151)) (e.g., via the JSON-RPC API), the message integrity validator won't be added to that topic, leaving those topics out of the claimed "DoS protection".

Also, dont see a problem. Having some validation is better than none at all. If api subs are not validated, then we have to add that, but not remove all validations. Better some than none.

> As they are executed on every message's arrival, validators must be fast and efficient to not impact the message latency and overall system performance

Sure, we will have two decodings, one for validating and one for the actual decoding. Can be improved, yes. But aren't we overoptimizing? Whats even worse is wrong messages being gossiped in the network. This do imply way more validations. For inspiration we can use [nimbus](https://github.com/status-im/nimbus-eth2/blob/ba7c0bc091161f261148dbfcb27b21cc48f1fdf8/beacon_chain/networking/eth2_network.nim#L2423).

**tldr**: i would leave this validation, as its better having some than none. unless im missing something i dont see anything critical here.

Reply:
I am not against the "claimed" DoS protection. I said that it was not the moment of merging it. And the way it was implemented, had flaws.

I commented it out because it uses the `defaultPubsubTopics` field. After this PR, this field is no longer available. So, the snippet cannot work. This will be included in one of the upcoming PRs: **once the validator implementation strategy is defined (in code)**.